app/controllers/references_controller.rb

```diff
@@ -9,6 +9,8 @@ def show
   def create
     list = List.find(reference_params[:list_id])
 
+    return redirect_to(:back, alert: @locator.errors.join(' ')) unless @locator.valid?
```

Comment:
dunno how the depracated process might affect this, but I wanted to bring up redirect_back in case you wanted to use it instead:

http://guides.rubyonrails.org/layouts_and_rendering.html#using-redirect-to